### ISSUE TITLE
The target websocket watch the specific url instead of upstream(host) with path

### ISSUE THREAD

**Issue by Dokki** [fastify/fastify-http-proxy#303]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the feature has not already been requested

### 🚀 Feature Proposal

Ability to customize the url for websockets, currently it is not possible.

Best regards, Anton.

### Motivation

I use **fastify-http-proxy** for development, When I run project then the proxy is look at some server, for example: test, stage, feature.
Example:
I have server with backend api url like this [https://api-test.myserver.com](https://api-test.myserver.com), and websocket url like this [https://websocket.myserver.com/conection](https://websocket.myserver.com/conection).
My local frontend looks at the proxy, and the proxy looks at backend api url. It working is well but when I turn on the **websocket: true** property then proxy trying to conect to [wss://api-test.myserver.com](https://api-test.myserver.com) and the url for sockets cannot be configured.
I ask to make it possible to configure the url for the websockets. Or can I suggest a solution with the pull request.

### Example

Basically the solution is simple. Need add new property to proxy options, like **wsUpstream**.
This code
```
if (options.upstream !== '') {
    wsProxy.addUpstream(fastify.prefix, rewritePrefix, options.upstream, options.wsClientOptions)
    // The else block is validate earlier in the code
  }
```
Replace with
```
if (isDefinedString(options.wsUpstream) || isDefinedString(options.upstream)) {
    wsProxy.addUpstream(
      fastify.prefix,
      rewritePrefix,
      options.wsUpstream ? options.wsUpstream : options.upstream,
      options.wsClientOptions,
    )
    // The else block is validate earlier in the code
  }
```
Helper function
```
function isDefinedString (string) {
  return typeof string === 'string' && !!string
}
```

**Comment by mcollina** [comment]
Seems legit! Why don't you send a PR?

**Comment by Dokki** [comment]
@mcollina Ok. Thx, I will prepare the PR soon.

**Comment by Dokki** [comment]
Waiting aproval for PR or feedback comments.

**Comment by Dokki** [comment]
Hey guys, сan I find out when the PRs will merge?

**Comment by Dokki** [comment]
@mcollina Can you rerun this job https://github.com/fastify/fastify-http-proxy/actions/runs/5188893141/jobs/9353133425
Smthg wrong with "Install dependencies" step.
Maybe:
1. Can't get access to folder.
2. Some files is locked with some process or processes.
3. Need run "npm cache clean".

Version 9.2.0 is not present in git becouse of failing job.

**Comment by Dokki** [comment]
@mcollina Hi there. I was create another PR with fix form my code from [there](https://github.com/fastify/fastify-http-proxy/pull/304).

This is my mistake, I would like to fix it. 
Can you check the PR when you have time?
PR: https://github.com/fastify/fastify-http-proxy/pull/308

Best regards, Anton.

**Comment by Dokki** [comment]
@mcollina Can you bump tag from 9.2.0 to 9.2.1?
Thx.
